sympy/core/expr.py

```diff
@@ -727,7 +729,10 @@ def equals(self, other, failing_expression=False):
         # don't worry about doing simplification steps one at a time
         # because if the expression ever goes to 0 then the subsequent
         # simplification steps that are done will be very fast.
-        diff = factor_terms(simplify(self - other), radical=True)
+        if self.has(TrigonometricFunction):
+            diff = factor_terms(trigsimp(self - other))
+        else:
+            diff = factor_terms(simplify(self - other), radical=True)
```

Comment:
```suggestion
            diff = trigsimp(self - other)
        else:
            diff = self - other
        diff = factor_terms(simplify(diff), radical=True)
```

Reply:
But maybe the better thing to do is to rewrite trig functions as exp. Try the following in your branch.
```
eq = 1/a - 1/b
eq.equals(0)
eq.rewrite(exp).equals(0)
```
I am not sure what all can rewrite in terms of exp, however, and whether we should do `diff.replace(lambda x: isinstance(x, TrigonometricFunction), lambda x: x.rewrite(exp))` or not.

Reply:
Yes, rewriting as `TrignometricFunction` works better here; previously `tan(3*x)` and `(3*tan(x) - tan(x)**3)/(1 - 3*tan(x)**2)` were not equal, but by using this it works. I've also added some tests for this.

```
a = tan(2*x)
b = 2*tan(x)/(1 - tan(x)**2)
eq = 1/a - 1/b
eq.equals(0)
eq.rewrite(exp).equals(0)
```
I've also tested this in my branch, and it works as expected. The output is `True`.